debian/assets/carlos_ctl/o19docs.py

```
@@ -0,0 +1,806 @@
+        o19import.report_append(state_dir, "P5 HRM relocation",
+                                "\n".join(hrm_lines))
+    # every pass, not only the first: an operator who fixed the tree by
+    # hand (the documented remedy) leaves root-owned files behind, and a
+    # root-run reconciliation would never notice
+    apply_ownership(ctx_root, ctx["dev_target"])
+    # two rows reaching one basename through different paths would be
+    # folded onto one file by the rewrite: refused before it runs
+    twins = query(hrm_basename_twins_sql(ctx["target_db"]))
+    if twins:
+        private(["HRM report basenames reached through more than one "
+                 "path (the basename rewrite cannot tell them apart):"]
```

Ruff (0.16.3)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)